wp-content/plugins/core/src/Object_Meta/Post_Archive_Settings.php

```diff
@@ -0,0 +1,41 @@
+<?php declare(strict_types=1);
+
+namespace Tribe\Project\Object_Meta;
+
+use Tribe\Libs\ACF;
+
+class Post_Archive_Settings extends ACF\ACF_Meta_Group {
+	public const NAME = 'post_archive_settings';
+
+	public const TITLE = 'title';
+	public const DESCRIPTION = 'description';
+	public const HERO_IMAGE = 'hero_image';
+
+	public function get_keys(): array {
+		return [
+			self::TITLE,
+			self::DESCRIPTION,
+			self::HERO_IMAGE,
+		];
+	}
+
+	public function get_group_config(): array {
+		$group = new ACF\Group( self::NAME, $this->object_types );
+		$group->set( 'title', __( 'Post Archive Settings', 'tribe' ) );
+		$group->add_field( $this->get_field( __( 'Archive Title', 'tribe' ), self::TITLE, 'text' ) );
+		$group->add_field( $this->get_field( __( 'Description', 'tribe' ), self::DESCRIPTION, 'textarea' ) );
+		$group->add_field( $this->get_field( __( 'Hero Image', 'tribe' ), self::HERO_IMAGE, 'image' ) );
+
+		return $group->get_attributes();
+	}
+
+	private function get_field( string $label, string $name, string $type ): ACF\Field {
```

Comment:
I've seen [several](https://github.com/moderntribe/square-one/blob/d60ead653a5c1386d78777609580ef0899ef3844/wp-content/plugins/core/src/Object_Meta/Social_Settings.php#L62-L71) other [examples](https://github.com/moderntribe/square-one/blob/d60ead653a5c1386d78777609580ef0899ef3844/wp-content/plugins/core/src/Object_Meta/Analytics_Settings.php#L32-L42) of this happening. I like the idea of only doing this once instead of everyone creating their own implementation, however, by abstracting it out, you'd make it harder to read/understand what's happening.

Setting up Meta fields is already very "magical" 🧙🏼 but I can be persuaded.

Reply:
Aren't we also losing translatable strings here?

Reply:
They are passed in using translation wrappers, so should pass through fine.

Reply:
I mention this mostly as I fine it somewhat un-readable when these configs get big. But agreed, an abstraction also just moves it farther away from the end user for understanding.

Reply:
Oh whoops, totally missed that. Introducing this method would be a decent use for a trait

Reply:
I've done this my self a few times on past projects. I agree, it is easier to read. And, it makes sense from a DRY standpoint.

That being said, I feel it's too far abstracted because of the reasons mentioned (differences in field types, required versus optional vals, etc). When it comes to modifying the implementation in a project (or just in the future), I have to make sure that the helper function (or trait) supports the a different format, key, or return value which increases the dev time versus just adding a new field registration to the end of the chain.